packages/salesforce/src/Adaptor.js

```diff
@@ -395,13 +328,111 @@ export function bulk(sObject, operation, options, fun) {
   };
 }
 
+const defaultBulkOptions = {
+  failOnError: true,
+  allowNoOp: true,
+  pollInterval: 6000,
+  pollTimeout: 240000,
+};
+/**
+ * Create and execute a bulk job using Salesforce Bulk API v2.
+ * @public
+ * @example
+ * bulk2(
+ *   "Patient__c",
+ *   "insert",
+ *   { failOnError: true, pollInterval: 3000, pollTimeout: 240000 },
+ *   (state) => {
+ *     return state.someArray.map((x) => ({ Age__c: x.age, Name: x.name }));
+ *   }
+ * );
+ * @function
+ * @param {String} sObject - API name of the sObject.
+ * @param {String} operation - The bulk operation to be performed
+ * @param {Object} options - Options passed to the bulk api.
+ * @param {Function} fun - A function which takes state and returns an array.
+ * @returns {Operation}
+ */
+export function bulk2(sObject, operation, options, fun) {
+  return async state => {
+    const { connection } = state;
+    const { failOnError, allowNoOp, pollTimeout, pollInterval } = {
```

Comment:
Bit of a wider issue this.... but this defaulting pattern has been spreading through the codebase. I really don't like it :( It always feels like a heavyweight solution to a simple problem.

I think it's time to put a stop to it.

Instead of the separate variable, the destructure and the spread, we can just do this:
```
const {
  failOnError = true,
  allowNoOp = true,
  pollInterval = 6000,
  pollTimeout = 240000,
  }  = options;
```

Reply:
I think what you have laid out is a good pattern, but that doesn't work if in a job code, a user does not use the optional params for example `pollInterval`.

For example my job code 
```
bulk2(
  'Some_sObject'
  'upsert', 
  {
    extIdField: 'External_ID',
    failOnError: true,
    allowNoOp: true,
  },
  state => state.data
);
```
Produce this error 
```
TypeError: Cannot set properties of undefined (setting 'pollInterval')
```

Reply:
Hmm, pretty suspicious of that. There's no reason it shouldn't work in the adaptor and runtime, and indeed it seems to work for me in CLI. Can we pair on this?